modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java

```diff
@@ -326,6 +311,14 @@ protected void initChannel(SocketChannel ch) throws Exception {
         serverBootstraps.put(name, serverBootstrap);
     }
 
+    protected ChannelHandler getServerChannelInitializer(String name, Settings settings) {
```

Comment:
Discussed this with @jasontedor, he explained that this class can be extended and these methods need to be overridable.